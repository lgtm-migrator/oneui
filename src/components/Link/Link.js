import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import { CONTEXTS } from '../../constants';
import styles from './Link.scss';

const { block } = bem({
    name: 'Link',
    classnames: styles,
    propsToMods: ['context']
});

const Link = props => {
    const { children, context, ...rest } = props;
    return (
        <a {...rest} {...block(props)}>
            {children}
        </a>
    );
};

Link.displayName = 'Link';

Link.propTypes = {
    /** The spinner context (e.g. brand, primary, bad, good etc.) */
    context: PropTypes.oneOf(['default', 'muted', ...CONTEXTS]),
    /** Link content */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired
};

Link.defaultProps = {
    context: 'default'
};

export default Link;
