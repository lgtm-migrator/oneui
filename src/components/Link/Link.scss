.Link {
    color: var(--link-color-normal);
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
        weight: var(--font-weight-normal);
    }
    line-height: var(--line-height-normal);
    text-decoration: var(--link-decoration-normal);
    transition: color var(--transition-duration);

    &:hover {
        color: var(--link-color-hover);
        text-decoration: var(--link-decoration-hover);
    }

    &--context {
        @each $context in (brand, neutral, primary, accent, info, good, warning, bad) {
            &_#{ $context } {
                color: var(--color-#{$context});

                &:hover {
                    color: var(--color-#{$context}-40);
                }
            }
        }

        &_muted {
            color: var(--color-muted);

            &:hover {
                color: var(--color-neutral);
            }
        }
    }
}
