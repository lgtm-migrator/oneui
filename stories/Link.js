import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, text, withKnobs } from '@storybook/addon-knobs';
import { Link } from '@textkernel/oneui';
import { CONTEXTS } from '@textkernel/oneui/constants';

storiesOf('Link', module)
    .addDecorator(withKnobs)
    .add('Link', () => (
        <Link
            target="_blank"
            href="https://textkernel.com"
            context={select('Context', ['default', 'muted', ...CONTEXTS], 'default')}
        >
            {text('Link text', 'Click me')}
        </Link>
    ));
