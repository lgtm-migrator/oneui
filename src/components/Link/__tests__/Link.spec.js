import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import Link from '../Link';

describe('<Link> that renders a link', () => {
    const hrefText = 'https://textkernel.com';
    const targetText = '_blank';

    it('should render default link correctly', () => {
        const { container } = render(<Link href={hrefText}>Click me</Link>);
        expect(container).toMatchSnapshot();
    });
    it('should add string html attributes correctly', () => {
        const { container } = render(
            <Link href={hrefText} target={targetText}>
                Click me
            </Link>
        );
        expect(container.querySelector('a')).toHaveAttribute('href', hrefText);
        expect(container.querySelector('a')).toHaveAttribute('target', targetText);
    });
});
