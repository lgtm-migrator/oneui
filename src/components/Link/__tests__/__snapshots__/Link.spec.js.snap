// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> that renders a link should render default link correctly 1`] = `
<div>
  <a
    class="Link"
    href="https://textkernel.com"
  >
    Click me
  </a>
</div>
`;
